### PR TITLE
bugfix local $this->allowed_types

### PR DESCRIPTION
**Pull request by w-huber**
Local field setting with setAllowedTypes() are always overwritten.

### CODE REVIEW

File: src/models/Link.php [hunk 1]
```diff
@@ -517,8 +517,10 @@ public function getTypes()
         if ($this->allowed_types) {
             // Prioritise local field over global settings
             $allowed_types = $this->allowed_types;
+        } else {
+            $allowed_types = $this->config()->get('allowed_types');
         }
-        if ($allowed_types = $this->config()->get('allowed_types')) {
+        if ($allowed_types) {
```
Comment (elliot-sawyer) [inline src/models/Link.php:500]: There's a potential scope issue here where $allowed_types is evaluated before it is set.  Try this instead:
```php
            ...
        $allowed_types = $this->config()->get('allowed_types');
        if ($this->allowed_types) {
            // Prioritise local field over global settings
            $allowed_types = $this->allowed_types;
        }
        if ($allowed_types) {
            ...
```

**Review by elliot-sawyer — CHANGES_REQUESTED** [review]

File: src/models/Link.php (hunk 1 above)
Comment (w-huber) [reply src/models/Link.php:500]: That's right. I resolved the issue.

**Comment by w-huber** [comment]
Is it possible to import this patch?

**Comment by elliot-sawyer** [comment]
Ah yep, sorry for the wait

**Pull request merged**